src/main/java/org/spongepowered/api/world/gen/WorldGeneratorProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.world.gen;
+
+import org.spongepowered.api.plugin.Plugin;
+
+/**
+ * If a plugin wishes to provide a world generator, the plugin class (the class
```

Comment:
I'm afraid that if an event is fired, every generator plugin installed will try to claim every world. I guess that most plugins will offer a setting to enable/disable the plugin for a world, but then the server admin has to find this setting for every world generator plugin.

This implementation allows the server admin to go to the WorldConfig and say: this world is assigned to AwesomeGenerator, that world to IceWorldGenerator, etc.

This implementation also makes it easy for Sponge to see which plugins are able to generate worlds (just an instanceof check), so that a nice generator selector can be shown in the singleplayer world creation menu.

Not that this implementation is ideal: it restricts each plugin to one generator. Maybe a similar system to command registration can be used for world generators?